src/store/commit.js

```diff
@@ -0,0 +1,33 @@
+import { indexGenerator } from "../api/fetch";
+
+export function fetchAll({ commit, rootState }, { collection, commitAction }) {
+  return new Promise((resolve, reject) => {
+    let i = 1;
+    let results = {};
+    const generator = indexGenerator(collection, rootState.auth);
+    commit("setStartLoading");
+    async function fetch() {
+      try {
+        let { value, done } = await generator.next();
+        for (let obj of value) {
+          results[obj.id] = obj;
+        }
+        if (done) {
+          commit(commitAction, results);
+          commit("removeOld");
+          resolve(true);
+        } else {
+          if (i % 5 === 0) {
+            commit(commitAction, results);
+            results = {};
+          }
+          i++;
+          fetch();
+        }
+      } catch (error) {
+        reject(error);
+      }
+    }
+    fetch();
```

Comment:
If `fetch` is an async function, you need to treat it as such. Either `await` it, or call `.then` on the implicit Promise it returns. (But like I said, making `fetchAll` async and `await`ing here feels more right to me.)